editor/src/components/navigator/left-pane.tsx

```diff
@@ -1376,21 +1416,114 @@ const GithubPane = React.memo(() => {
   )
 })
 
-const FileChanges = ({ files, type }: { files: string[]; type: GithubFileStatus }) => {
-  if (files.length === 0) {
+const Ellipsise = ({
+  children,
+  title,
+  style,
+}: {
+  children: any
+  title?: string
+  style?: React.CSSProperties
+}) => {
+  return (
+    <div
+      style={{
+        ...style,
+        whiteSpace: 'nowrap',
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+      }}
+      title={title}
+    >
+      {children}
+    </div>
+  )
+}
+
+const handleClickRevertAllFiles = (dispatch: EditorDispatch) => (e: React.MouseEvent) => {
```

Comment:
These should be defined in the component where they're used wrapped in `React.useCallback` to avoid lots of instances being created as a result of re-renders.

Reply:
https://github.com/concrete-utopia/utopia/pull/2743/commits/7f64023515b2de51952acdd1a0e80747d9c48019